doc/theming.rst

```diff
@@ -310,6 +290,44 @@ Python :mod:`ConfigParser` module) and has the following structure:
   and are accessible from all templates as ``theme_<name>``.
 
 
+.. _distribute-your-theme:
+
+Distribute your theme as a python package
+-----------------------------------------
+
+As a way to distribute your theme, you can use python package.  Python package
+brings to users easy setting up ways.
+
+To distribute your theme as a python package, please define an entry point
+called ``sphinx.html_themes`` in your setup.py file, and write a ``setup()``
+function to register your themes in it::
+
+    # 'setup.py'
+    setup(
+        ...
+        entry_points = {
+            'sphinx.html_themes': [
+                'name_of_theme = your_package',
+            ]
+        },
+        ...
+    )
+
+    # 'your_package.py'
+    from os import path
+
+    def setup(app):
+        app.add_html_theme('name_of_theme', path.abspath(path.dirname(__file__)))
```

Comment:
Is name needed?
If theme extension have multiple theme directory in single parent `theme_path`, the extension will call `app.add_html_theme(...)` for each theme **name** with one **theme_path**. Is it OK as a usage?

Additionally, in such case, does user need specify `name_of_theme` for each themes for `app.add_html_theme` and `sphinx.html_themes` ?

I think multiple themes in single package is not common case, however it is better to cover such case in this document.